src/unistd.rs

```diff
@@ -2136,3 +2146,424 @@ mod setres {
+                    unsafe {
+                        libc::getpwuid_r(uid, &mut pwd,
+                                         cbuf.as_mut_ptr(),
+                                         cbuf.len(), &mut res) 
+                    }
+                },
+                UserQuery::NameWithBufsize(name, _) |
+                UserQuery::Name(name) => {
+                    unsafe {
+                        libc::getpwnam_r(CString::new(name).unwrap().as_ptr(), 
+                                         &mut pwd, cbuf.as_mut_ptr(), cbuf.len(), &mut res) 
+                    }
+                },
+            }
+
+        };
+
+        match i {
+            0 => {
+                if !res.is_null() {
+                    Some(Ok(User::from(res))) 
+                } else {
+                    None
+                }
+            },
+            _ => { Some(Err(Error::Sys(Errno::last()))) },
+        }
+    }
+}
+
+impl Queryable<GroupQuery> for Group {
+    fn query(q: GroupQuery) -> Option<Result<Self>> {
+        let mut cbuf = match q {
+            GroupQuery::GidWithBufsize(_, size) | 
+            GroupQuery::NameWithBufsize(_, size) => {
+                vec![0 as c_char; size]
+            },
+            _ => {
+                vec![0 as c_char; PWGRP_BUFSIZE]
+            }
+        };
+
+        let mut grp: libc::group =  unsafe { mem::zeroed() };
+        let mut res = ptr::null_mut();
+
+        let i = {
+            unsafe { Errno::clear(); }
+
+            match q {
+                GroupQuery::GidWithBufsize(Gid(gid), _) |
+                GroupQuery::Gid(Gid(gid)) => { 
+                    unsafe { 
+                        libc::getgrgid_r(gid, &mut grp,
+                                         cbuf.as_mut_ptr(), 
+                                         cbuf.len(), &mut res) 
+                    }
+                },
+                GroupQuery::NameWithBufsize(name, _) |
+                GroupQuery::Name(name) => {
+                    unsafe {
+                        libc::getgrnam_r(CString::new(name).unwrap().as_ptr(), 
+                                         &mut grp, cbuf.as_mut_ptr(), cbuf.len(), &mut res) 
+                    }
+                },
+            }
+
+        };
+
+        match i {
+            0 => {
+                if !res.is_null() {
+                    Some(Ok(Group::from(res))) 
+                } else {
+                    None
+                }
+            },
+            _ => { Some(Err(Error::Sys(Errno::last()))) },
+        }
+    }
+}
+
+#[cfg(not(any(target_os = "android", 
+              target_os = "ios", 
+              target_os = "macos", 
+              target_env = "musl",
+              target_arch = "mips",
+              target_arch = "mips64",
+              target_arch = "mips64el",
+              target_arch = "mipsel",
+              target_arch = "s390x")))] 
+mod usergroupiter {
+    use libc;
+    use libc::{c_char};
```

Comment:
You can just do `use libc::{self, c_char};`

Reply:
done.